src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -611,7 +602,7 @@ class Peer(
                             onChainFeeratesFlow.filterNotNull().first()
                         )
                         val channelConfig = ChannelConfig.standard
-                        // Note that we don't contribute yet to the channel funding (we need dual funding for that).
+                        // We currently don't add any inputs to the funding transaction.
                         val (state1, actions1) = state.process(ChannelEvent.InitNonInitiator(msg.temporaryChannelId, FundingInputs.empty, localParams, channelConfig, theirInit!!))
                         val (state2, actions2) = state1.process(ChannelEvent.MessageReceived(msg))
                         processActions(msg.temporaryChannelId, actions1 + actions2)
```

Comment:
Those lines need to be interverted, otherwise `processActions(ChannelAction.ChannelId.IdAssigned` is a no-op because the `_channels` map is empty:
```scala
_channels = _channels + (msg.temporaryChannelId to state2)
processActions(msg.temporaryChannelId, actions1 + actions2)
```

Reply:
We're definitely missing a test here, I'll fix that and add one when I rebase!

Reply:
There is another related issue. The code below (which was in the `HasTemporaryChannelId`) needs to be executed to clean up the old binding in the `_channels` map:

```scala
actions.filterIsInstance<ChannelAction.ChannelId.IdAssigned>().forEach {
  _channels = _channels - it.temporaryChannelId + (it.channelId to state1)
}
```